backend/src/main/java/fk/prof/backend/worker/BackendDaemon.java

```diff
@@ -0,0 +1,198 @@
+
+    this.leaderReadContext = leaderReadContext;
+    this.processGroupAssociationStore = processGroupAssociationStore;
+    this.aggregationWindowLookupStore = aggregationWindowLookupStore;
+    this.simultaneousWorkAssignmentCounter = simultaneousWorkAssignmentCounter;
+  }
+
+  @Override
+  public void start() {
+    httpClient = buildHttpClient();
+    aggregationWindowPlannerStore = buildAggregationWindowPlannerStore();
+    postLoadToLeader();
+  }
+
+  private ProfHttpClient buildHttpClient() {
+    JsonObject httpClientConfig = configManager.getHttpClientConfig();
+    ProfHttpClient httpClient = ProfHttpClient.newBuilder()
+        .keepAlive(httpClientConfig.getBoolean("keepalive", false))
+        .useCompression(httpClientConfig.getBoolean("compression", true))
+        .setConnectTimeoutInMs(httpClientConfig.getInteger("connect.timeout.ms", 2000))
+        .setIdleTimeoutInSeconds(httpClientConfig.getInteger("idle.timeout.secs", 3))
+        .setMaxAttempts(httpClientConfig.getInteger("max.attempts", 1))
+        .build(vertx);
+    return httpClient;
+  }
+
+  private AggregationWindowPlannerStore buildAggregationWindowPlannerStore() {
+    return new AggregationWindowPlannerStore(
+        vertx,
+        config().getInteger("aggregation.window.duration.mins", 30),
+        config().getInteger("aggregation.window.end.tolerance.secs", 120),
+        config().getInteger("workprofile.refresh.offset.secs", 300),
+        config().getInteger("scheduling.buffer.secs", 30),
+        config().getInteger("work.assignment.max.delay.secs", 120),
+        simultaneousWorkAssignmentCounter,
+        aggregationWindowLookupStore,
+        this::getWorkFromLeader);
+  }
+
+  private void postLoadToLeader() {
+    String leaderIPAddress;
+    if((leaderIPAddress = leaderReadContext.getLeaderIPAddress()) != null) {
+
+      //TODO: load = 0.5 hard-coded right now. Replace this with dynamic load computation in future
+      float load = 0.5f;
+
+      try {
+        httpClient.requestAsync(
+            HttpMethod.POST,
+            leaderIPAddress,
+            leaderHttpPort,
+            ApiPathConstants.LEADER_POST_LOAD,
+            ProtoUtil.buildBufferFromProto(
+                BackendDTO.LoadReportRequest.newBuilder()
+                    .setIp(ipAddress)
+                    .setPort(backendHttpPort)
+                    .setLoad(load)
+                    .setCurrTick(loadTickCounter++)
+                    .build()))
+            .setHandler(ar -> {
+              if(ar.succeeded()) {
+                if(ar.result().getStatusCode() == 200) {
+                  try {
+                    Recorder.ProcessGroups assignedProcessGroups = ProtoUtil.buildProtoFromBuffer(Recorder.ProcessGroups.parser(), ar.result().getResponse());
+                    processGroupAssociationStore.updateProcessGroupAssociations(assignedProcessGroups, (processGroupDetail, processGroupAssociationResult) -> {
```

Comment:
Fixed in commit https://github.com/Flipkart/fk-prof/commit/4082bc7ca78e11f75aa121c72aef00de8f64c539